src/Connect.js

```diff
@@ -45,24 +45,26 @@ class Connect {
     if (this.network === network.defaults.networks.mainnet && this.accountType === 'segregated') {
       throw new Error('Segregated accounts are not supported on mainnet')
     }
-
-    // State
-    this.did = null
-    this.mnid = null
-    this.address = null
-    this.doc = null
-    this.pushToken = null
-    this.publicEncKey = null
+  
+    // Initialize private state
+    this._state = {
+      did: null, 
+      mnid: null, 
+      address: null, 
+      doc: null, 
+      pushToken: null, 
+      publicEncKey: null
+    }
 
     // Load any existing state if any
-    if (this.storage)  this.getState()
-    if (!this.keypair) this.keypair = Credentials.createIdentity()
+    if (this.storage) this.getState()
+    if (!this.keypair || !this.keypair.did) this.keypair = Credentials.createIdentity()
```

Comment:
forgot is this condition for any case `!this.keypair.did`?

Reply:
This is just because `get this.keypair` returns `{...this._state.keypair}`. If `this._state.keypair` is null or undefined, `this.keypair` still returns `{}` which is truthy-- I wasn't sure of a more elegant solution for this, except maybe just removing the `!this.keypair` as it will always be false